R/pip.R

```diff
@@ -14,12 +14,13 @@ pip_version <- function(python) {
 }
 
 
-pip_install <- function(python, packages, ignore_installed = FALSE) {
+pip_install <- function(python, packages, pip_options = character(), ignore_installed = FALSE) {
 
   # construct command line arguments
   args <- c("-m", "pip", "install", "--upgrade")
   if (ignore_installed)
     args <- c(args, "--ignore-installed")
+  args <- c(args, pip_options)
```

Comment:
Some questions:

1. Should this be `pip_arguments` rather than `pip_options`? Since this generically allows any kind of argument to `pip`, not just `--install-option` as described in the issue.

2. Do we need to `shQuote()` these arguments? Or do you think the onus is on the caller to shell-quote arguments as necessary?

Reply:
1. I don't have opinion about the name, but I remember `devtools` has used name like `build_opts` for extra command line arguments passed to `R CMD build`. So `pip_options` possibly should be fine.

2. That's a good point. But I think if we `shQuote` here, it will force user to use `c("--option1", "--option2")` instead of `"--option1 --option2"`. Also I checked what `devtools` do with the `build_opts`, they do not `shQuote` them. Possibly we can follow that.

Reply:
Both good points. Okay, I'm happy with this as-is. Thanks!